docs/labels.md

```diff
@@ -25,14 +25,16 @@ Labels propagate among Tekton entities as follows:
 
 - For `Pipelines` instantiated using a `PipelineRun`, labels propagate
 automatically from `Pipelines` to `PipelineRuns` to `TaskRuns`, and then to
-the associated `Pods`.
+the associated `Pods`. If a label is present in both `Pipeline` and
+`PipelineRun`, the label in `PipelineRun` takes precedence.
 
 - Labels from `Tasks` referenced by `TaskRuns` within a `PipelineRun` propagate to the corresponding `TaskRuns`,
-and then to the associated `Pods`.
+and then to the associated `Pods`. As for `Pipeline` and `PipelineRun`, if a label is present in both `Task` and
+`TaskRun`, the label in `TaskRun` takes precedence.
 
 - For standalone `TaskRuns` (that is, ones not executing as part of a `Pipeline`), labels
 propagate from the [referenced `Task`](taskruns.md#specifying-the-target-task), if one exists, to
-the corresponding `TaskRun`, and then to the associated `Pod`.
+the corresponding `TaskRun`, and then to the associated `Pod`. The same as above applies.
 
 - For `Conditions`, labels propagate to the corresponding `TaskRuns`, and then to the associated `Pods`.
```

Comment:
We don't have a doc for the same behavior on labels, should we rewrite this one to be labels and annotations ?
/cc @pritidesai @afrittoli @bobcatfish

Reply:
yup indeed, lets be specific and document labels and annotations both. Also the doc can be updated to include an example!